### PR TITLE
Implement acl models

### PR DESCRIPTION
**Pull request by ronaldemanuel**
# Main changes
* Created main ACL `models`:
    * Permissions
    * Roles
* Implemented your `migrations`.
* Defined the default `seeders` with predefined **roles** and **permissions**.

### CODE REVIEW

File: app/Models/Permission.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export default class Permission extends BaseModel {
+  @column({ isPrimary: true })
+  public id: number
+
+  @column()
+  public name: string
+
+  @column()
+  public description: string
+
```
Comment (Manuel-Antunes) [inline app/Models/Permission.ts:13]: please, add "slug" field

File: app/Models/Role.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { DateTime } from 'luxon'
+import { BaseModel, column } from '@ioc:Adonis/Lucid/Orm'
+
+export default class Role extends BaseModel {
+  @column({ isPrimary: true })
+  public id: number
+
+  @column()
+  public name: string
```
Comment (Manuel-Antunes) [inline app/Models/Role.ts:9]: please, add "slug" field

File: database/migrations/1630111942031_permissions.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import BaseSchema from '@ioc:Adonis/Lucid/Schema'
+
+export default class Permissions extends BaseSchema {
+  protected tableName = 'permissions'
+
+  public async up() {
+    this.schema.createTable(this.tableName, (table) => {
+      table.increments('id').primary()
+      table.string('name').notNullable().unique()
+      table.text('description').notNullable()
```
Comment (Manuel-Antunes) [inline database/migrations/1630111942031_permissions.ts:10]: please, add "slug" field

File: database/migrations/1630114185334_roles.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import BaseSchema from '@ioc:Adonis/Lucid/Schema'
+
+export default class Roles extends BaseSchema {
+  protected tableName = 'roles'
+
+  public async up() {
+    this.schema.createTable(this.tableName, (table) => {
+      table.increments('id').primary()
+      table.string('name').notNullable().unique()
+      table.text('description').notNullable()
```
Comment (Manuel-Antunes) [inline database/migrations/1630114185334_roles.ts:10]: please, add "slug" field

File: database/seeders/Permission.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import BaseSeeder from '@ioc:Adonis/Lucid/Seeder'
+import Permission from 'App/Models/Permission'
+
+export default class PermissionSeeder extends BaseSeeder {
+  public async run() {
+    await Permission.createMany([
+      {
+        name: 'read_self',
```
Comment (Manuel-Antunes) [inline database/seeders/Permission.ts:8]: please, add "slug" field

File: database/seeders/Role.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  public async run() {
+    await Role.createMany([
+      {
+        name: 'global',
+        description: 'Has all permissions',
+      },
+      {
+        name: 'admin',
+        description:
+          'Has common user creation permissions. Reading, updating and deleting both the common and the same',
```
Comment (Manuel-Antunes) [inline database/seeders/Role.ts:14]: please, add "slug" field

**Review by Manuel-Antunes — CHANGES_REQUESTED** [review]
Add the field "slug" as a string in roles and permissions

**Comment by ronaldemanuel** [comment]
I added the **slug** fields on models and migrations and their respective values ​​in seeders

**Review by Manuel-Antunes — APPROVED** [review]
Nice!

**Pull request merged**